amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -111,16 +117,22 @@ public TaskResult call() {
 
                 final long startTime = System.currentTimeMillis();
                 if (reason == ShutdownReason.SHARD_END) {
-                    // Create new lease for the child shards if they don't exist.
-                    // We have one valid scenario that shutdown task got created with SHARD_END reason and an empty list of childShards.
-                    // This would happen when KinesisDataFetcher(for polling mode) or FanOutRecordsPublisher(for StoS mode) catches ResourceNotFound exception.
-                    // In this case, KinesisDataFetcher and FanOutRecordsPublisher will send out SHARD_END signal to trigger a shutdown task with empty list of childShards.
-                    // This scenario could happen when customer deletes the stream while leaving the KCL application running.
-                    if (!CollectionUtils.isNullOrEmpty(childShards)) {
-                        createLeasesForChildShardsIfNotExist();
-                        updateLeasesForChildShards();
+                    final Lease currentShardLease = leaseCoordinator.getCurrentlyHeldLease(leaseKeyProvider.apply(shardInfo));
```

Comment:
Originally didn't use the cached lease since it wouldn't have the child shard information, but we can change the return type of UpdateLeaseWithChildShards to return the modified lease.